03_LabJat/letterFreequencyCount_test.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLettersOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := " :3\nS:1\nT:1\na:1\ng:1\nh:1\ni:3\nn:1\nr:1\ns:2\nt:1\n"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestLettersLength(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	m := letters("aaa")
+
+	// Then
+	expected := 1
+	actual := len(m)
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestLettersOccurance(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	m := letters("aaa")
+
+	// Then
+	expected := 3
+	actual := m['a']
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestSortedLetters(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	m := letters("abaa")
+	k := sortedLetters(m)
```

Comment:
This test can fail if the `letters()` function breaks, because it is relying on it working properly. Ideally test cases are kept small and test one thing only. Since `sortedLetters()` takes a map as input, you should specify that map in this test case so you don't need to call `letters()`.

```
    k := sortedLetters(map[rune]int{'a': 3, 'b': 1})
```